BadgeUpClient/ResourceClients/MetricClient.cs

```diff
@@ -24,5 +25,30 @@ public MetricClient(BadgeUpHttpClient httpClient)
 		{
 			return await this.m_httpClient.Get<MetricResponse>(ENDPOINT + "/" + subjectId + "/" + key);
 		}
+
+
+		/// <summary>
+		/// Retrieves a list of all metrics.
+		/// </summary>
+		/// <returns></returns>
+		public async Task<List<MetricResponse>> GetAll()
+		{
+			var result = new List<MetricResponse>();
+			string path = "/v1/apps/{applicationId}";
+
+			string url = ENDPOINT;
+
+			do
+			{
+				var response = await this.m_httpClient.Get<MetricMultipleResponse>(url, path);
```

Comment:
Can we make the Get All iterator functionality a common part of the BadgeUp HTTP client? I'd like to share it across different resource clients.